### PR TITLE
move Provider base trait to goose-providers

### PR DESCRIPTION
**Pull request by jamadeo**
Moves the Provider trait to goose-providers. Re-exports because many files importing this will be moved themselves, so we can avoid some churn of changing `use` lines.

Related to #9803

### CODE REVIEW

File: crates/goose-providers/src/base.rs [hunk 1]
```diff
@@ -1,17 +1,563 @@
...
+    async fn complete(
+        &self,
+        model_config: &ModelConfig,
+        session_id: &str,
+        system: &str,
+        messages: &[Message],
+        tools: &[Tool],
+    ) -> Result<(Message, ProviderUsage), ProviderError> {
+        let stream = self
+            .stream(model_config, session_id, system, messages, tools)
```
[the remark this answers is not in the document]
Comment (jamadeo) [reply crates/goose-providers/src/base.rs:205]: not sure what this is talking about - the prior version also forwards model_config

File: crates/goose/src/config/mod.rs [hunk 2]
```diff
@@ -21,7 +20,7 @@ pub use extensions::{
     get_extension_by_name, get_warnings, is_extension_enabled, remove_extension,
     resolve_extensions_for_new_session, set_extension, set_extension_enabled, ExtensionEntry,
 };
-pub use goose_mode::GooseMode;
+pub use goose_providers::goose_mode::GooseMode;
```
[the remark this answers is not in the document]
Comment (jamadeo) [reply crates/goose/src/config/mod.rs:23]: this is fine

File: crates/goose/src/permission/mod.rs [hunk 3]
```diff
@@ -1,8 +1,10 @@
-pub mod permission_confirmation;
 pub mod permission_inspector;
 pub mod permission_judge;
 pub mod permission_store;
 
-pub use permission_confirmation::{Permission, PermissionConfirmation};
+pub use goose_providers::permission::{Permission, PermissionConfirmation};
+pub mod permission_confirmation {
+    pub use goose_providers::permission::PrincipalType;
```
[the remark this answers is not in the document]
Comment (jamadeo) [reply crates/goose/src/permission/mod.rs:7]: let the compiler decide!

**Review by DOsinga — APPROVED** [review]
Clean mechanical refactor — extracting session-naming logic off the `Provider` trait into a standalone `session_naming` module and moving the trait into `goose-providers` with re-export shims to avoid import churn. Verified the moved code matches the originals and the session_naming tests are genuine (unicode/attribute/orphan-tag edge cases). Codex's three comments are all addressed; the "thinking effort" one is a false positive — the old `complete` path forwards `model_config` unchanged too. LGTM.
